proxy/router/src/lib.rs

```diff
@@ -73,8 +71,13 @@ where T: Recognize,
 struct Inner<T>
 where T: Recognize,
 {
-    routes: IndexMap<T::Key, T::Service>,
     recognize: T,
+    cache: Mutex<Cache<T::Key, T::Service>>,
```

Comment:
If we are nit picking, why not move `Mutex` into `Cache`, then the lock is fully encapsulated and you could remove it from there later.

This is not a critical change, just a thought.

Reply:
Then you'd need yet another struct, since there's multiple fields in `Cache` that need to be accessed behind the lock..

Reply:
really, only the map needs to be locked. i'll make this change with the followup Cache change https://github.com/runconduit/conduit/pull/920